api/v1/_email/PKPEmailHandler.inc.php

```diff
@@ -0,0 +1,210 @@
+					'handler' => [$this, 'create'],
+					'roles' => [ROLE_ID_SITE_ADMIN, ROLE_ID_MANAGER],
+				],
+			],
+			'PUT' => [
+				[
+					'pattern' => $this->getEndpointPattern() . '/{queueId}',
+					'handler' => [$this, 'process'],
+					'roles' => [ROLE_ID_SITE_ADMIN, ROLE_ID_MANAGER],
+				],
+			],
+		];
+		parent::__construct();
+	}
+
+	/**
+	 * @copydoc PKPHandler::authorize
+	 */
+	public function authorize($request, &$args, $roleAssignments) {
+		import('lib.pkp.classes.security.authorization.PolicySet');
+		$rolePolicy = new PolicySet(COMBINING_PERMIT_OVERRIDES);
+
+		import('lib.pkp.classes.security.authorization.RoleBasedHandlerOperationPolicy');
+		foreach ($roleAssignments as $role => $operations) {
+			$rolePolicy->addPolicy(new RoleBasedHandlerOperationPolicy($request, $role, $operations));
+		}
+		$this->addPolicy($rolePolicy);
+
+		return parent::authorize($request, $args, $roleAssignments);
+	}
+
+	/**
+	 * Create a jobs queue to send a bulk email to users in one or
+	 * more user groups
+	 *
+	 * @param ServerRequestInterface $slimRequest
+	 * @param APIResponse $response
+	 * @param array $args arguments
+	 * @return APIResponse
+	 */
+	public function create(ServerRequestInterface $slimRequest, APIResponse $response, array $args) {
+		$context = $this->getRequest()->getContext();
+		$contextId = $context->getId();
+
+		$requestParams = $slimRequest->getParsedBody();
+
+		$params = [];
+		foreach ($requestParams as $param => $val) {
+			switch ($param) {
+				case 'userGroupIds':
+					if (!is_array($val)) {
+						$val = strlen(trim($val))
+							? explode(',', $val)
+							: [];
+					}
+					$params[$param] = array_map('intval', $val);
+					break;
+				case 'body':
+				case 'subject':
+					$params[$param] = $val;
+					break;
+			}
+		}
+
+		$errors = [];
+		if (empty($params['body'])) {
+			$errors['body'] = [__('api.emails.400.missingBody')];
+		}
+
+		if (empty($params['subject'])) {
+			$errors['subject'] = [__('api.emails.400.missingSubject')];
+		}
+
+		if (empty($params['userGroupIds'])) {
+			$errors['userGroupIds'] = [__('api.emails.400.missingUserGroups')];
+		}
+
+		if ($errors) {
+			return $response->withJson($errors, 400);
+		}
+
+		$userGroupDao = DAORegistry::getDAO('UserGroupDAO');
+		foreach ($params['userGroupIds'] as $userGroupId) {
+			if (!$userGroupDao->contextHasGroup($contextId, $userGroupId)) {
+				return $response->withJson([
+						'userGroupIds' => [__('api.emails.403.notAllowedUserGroup')],
+					], 403);
+			}
+		}
+
+		// Only permit emails to be sent to active users in this context
+		$params['status'] = 'active';
+		$params['contextId'] = $contextId;
+
+		$userIds = Services::get('user')->getIds($params);
+		$subject = $params['subject'];
+		$body = $params['body'];
+		$fromEmail = $context->getData('contactName');
+		$fromName = $context->getData('contactEmail');
+		$queueId = 'email_' . uniqid();
+		$batches = array_chunk($userIds, self::EMAILS_PER_JOB);
```

Comment:
This one line basically replaces the `MassEmail` class, which is still lying around unused in `pkp-lib`. I'll remove it :)

Reply:
I've removed it.